### ISSUE TITLE
How to export a model with only batched NMS layer

### ISSUE THREAD

**Issue by 1224wxwx** [grimoire/mmdetection-to-tensorrt#48]
Hi, thanks for your amazing work!

I want to export a model with only "BatchedNMS" (located in post_processing/batched_nms.py) for some reason. In other words, I just want to speed up the process of NMS through tensorRT.
Could you please tell me how to organize the config file, or how to export it directly?

Thank you!

**Comment by grimoire** [comment]
Hi
If you want to convert [batched_nms.py](https://github.com/grimoire/mmdetection-to-tensorrt/blob/master/mmdet2trt/core/post_processing/batched_nms.py) in this repo, please try follow steps:
import this project to load the convertor
```python
import mmdet2trt
```

create the layer manually
```python
layer = mmdet2trt.core.post_processing.batched_nms.BatchedNMS(....)
```

convert the layer with torch2trt_dynamic
```python
torch2trt_dynamic(layer, dummy_input, ....)
```

**Comment by 1224wxwx** [comment]
> Hi
> If you want to convert [batched_nms.py](https://github.com/grimoire/mmdetection-to-tensorrt/blob/master/mmdet2trt/core/post_processing/batched_nms.py) in this repo, please try follow steps:
> import this project to load the convertor
> 
> ```python
> import mmdet2trt
> ```
> 
> create the layer manually
> 
> ```python
> layer = mmdet2trt.core.post_processing.batched_nms.BatchedNMS(....)
> ```
> 
> convert the layer with torch2trt_dynamic
> 
> ```python
> torch2trt_dynamic(layer, dummy_input, ....)
> ```

Thanks for your reply.

Do I need to warmup the model like what you did in function "mmdet2trt"？

Here's the quote:

`   logger.info("Model warmup")
    with torch.no_grad():
        result = wrap_model(dummy_input)`

**Comment by grimoire** [comment]
Nope.
I put the warmup code here to initial some values inside the module(Actually It is useless for now). Just convert the model directly.

**Comment by twmht** [comment]
@grimoire 

What did you modify for batched NMS layer? the plugin you provide is very similar to TensorRT(https://github.com/NVIDIA/TensorRT/tree/main/plugin/batchedNMSPlugin)?

**Comment by grimoire** [comment]
@twmht  I do not remember all details.
One important difference is that in [official implementation](https://github.com/NVIDIA/TensorRT/blob/0914fecb389294b902835fa7cd5a36b09ac0e310/plugin/common/kernels/allClassNMS.cu#L85) the offset is 1 for `intersect_width` and `intersect_height`. And in [my modification](https://github.com/grimoire/amirstan_plugin/blob/e92bd67c40dcca1ef111ac04ef029ddf23e96e58/src/plugin/batchedNMSPlugin/allClassNMS.cu#L69) it will be 0 so the nms will be aligned with the one in mmcv.
